### PR TITLE
big-o-notation - Simon Kim.

### PR DESCRIPTION
**Pull request by Skim-329**
Screen Shot 2021-11-15 at 4 19 31 PM

### CODE REVIEW

File: big-o-notation/big-o.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  const unique = []; // 1 * 1 = O(1)
+  for (
+    let i = 0; // 1 * 1 = O(1)
+    i < words.length; // 2 * n = O(2n)
+    i++ // 2 * n = O(2n)
+  ) {
+    const word = words[i]; // 1 * 1 = O(1)
+    let isUnique = true; // 1 * 1 = O(1)
+    for (
+      let c = 0; // 1 * 1 = O(1)
```
Comment (scbowler) [inline big-o-notation/big-o.js:31]: This will run once for each item in the array so this line will be `O(n)`

File: big-o-notation/big-o.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      c < i; // 2 * n = O(2n)
+      c++ // 2 * n = O(2n)
+    ) {
+      const comparing = words[c]; // 1 * 1 = O(1)
+      if (comparing === word) {
+        // 2 * n = O(2n)
+        isUnique = false; // 1 * 1 = O(1)
```
Comment (scbowler) [inline big-o-notation/big-o.js:38]: Since these lines are in a nested loop they will be `O(n^2)`

File: big-o-notation/big-o.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      }
+    }
+    if (isUnique) {
+      // 1 * 1 = O(1)
```
Comment (scbowler) [inline big-o-notation/big-o.js:42]: This is still inside the outer loop so it will run once for every item in the data set, so it will be `O(n)`

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good, but a few updates for the 2nd function. Let me know if you have any questions.

**Comment by Skim-329** [comment]
Screen Shot 2021-11-15 at 4 34 07 PM

**Comment by Skim-329** [comment]
Please disregard the previous pull request! Latest has changes and correct corrections :D 
Screen Shot 2021-11-15 at 5 32 14 PM

File: big-o-notation/big-o.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  const seen = {}; // 1 * 1 = O(1)
+  const unique = []; // 1 * 1 = O(1)
+  for (
+    let i = 0; // 1 * 1 = O(1)
+    i < words.length; // 2 * n = O(n)
+    i++ // 2 * n = O(n)
+  ) {
+    const word = words[i]; // 2 * n = O(n)
+    if (!seen[word]) {
+      // 3 * n = O(n)
```
Comment (TimelikeClosure) [inline big-o-notation/big-o.js:13]: Is this comment for the line above it or the line below it?

**Review by TimelikeClosure — APPROVED** [review]

File: big-o-notation/big-o.js (hunk 4 above)
Comment (Skim-329) [reply big-o-notation/big-o.js:13]: line above... I'm not sure why it reformatted that way

**Pull request merged**